ios/Classes/AudioRecorder.swift

```diff
@@ -157,4 +156,15 @@ public class AudioRecorder: NSObject, AVAudioRecorderDelegate{
             return Int(kAudioFormatMPEG4AAC)
         }
     }
+    
+    private func getDocumentDirectory(_ result: @escaping FlutterResult) -> String {
+        let directory = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
```

Comment:
NSSearchPathForDirectoriesInDomains can return a path that doesn't even contain the requested directory as stated [here](https://developer.apple.com/documentation/foundation/1414224-nssearchpathfordirectoriesindoma). Please use other method as suggested in "See Also" section of the attached link.